### PR TITLE
Add entity collisions

### PR DESCRIPTION
**Pull request by Chitasa**
Remade old pr to comply with updates
Added a new test for entity collisions

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -644,13 +681,37 @@ function getStatusEffectNamesForVersion (supportFeature) {
...
+
+  return new AABB(-w, 0, -w, w, entity.height, w).offset(pos.x, pos.y, pos.z)
+}
+
+function getIntersectingEntities (bb, entityList) {
+  const entities = []
+
+  for (let i = 0; i < entityList.length; i++) {
+    const e = entityList[i]
+    if (e.type !== 'object') {
```
Comment (iCollin) [inline index.js:696]:
```suggestion
    if (e.type !== 'object' && e.type !== 'orb') {
```

don't want to be pushed by `experience_orb`

Comment (U5B) [reply index.js:696]: A list of entities that the bot should not be pushed by should be there.

Or how about a property inside an entity marking it as pushable? That way people can set if they want their bot to be pushable by only certain entities

Comment (O-of) [reply index.js:696]: > Or how about a property inside an entity marking it as pushable? That way people can set if they want their bot to be pushable by only certain entities

https://github.com/PrismarineJS/prismarine-entity/blob/master/index.js#L24
I already committed an attribute for noclip, I guess I'll need to actually set it places 😅 

```suggestion
    if (!e.noClip && (e.type !== 'object' && e.type !== 'orb')) { // temp until I can set noclip flag in places
```

**Comment by rom1504** [comment]
would appreciate reviews

**Comment by rom1504** [comment]
@U5B maybe you'd like to take a look at this ?

**Review by U5B — APPROVED** [review]
LGTM (merge conflicts need to be fixed)

**Comment by U5B** [comment]
Another feature to this is to implement `collisionRule` from scoreboard.

**Comment by O-of** [comment]
I had to extract the physics object out of the Physics class so there's a placeholder value until it can be set when mcdata is defined. Is there any better way to do this or is this fine?

File: lib/features.json [hunk 2]
```diff
@@ -28,5 +28,10 @@
     "name": "effectNamesAreRegistryNames",
     "description": "Status effect names equal to their registry names",
     "versions": ["1.17"]
+  },
+  {
+    "name": "entityCollision",
+    "description": "Entities can collide with each other",
+    "versions": ["1.9", "1.10", "1.11", "1.12", "1.13", "1.14", "1.15", "1.16", "1.17"]
```
Comment (Robbilie) [inline lib/features.json:35]: needs 1.18 and probably a rebase

**Comment by Robbilie** [comment]
happy to test this once rebased @O-of :)
from what (little) i understand the code looks good to me too :P

**Comment by rom1504** [comment]
please finish

**Pull request closed without merging**